spec/puppet-server-lib/puppet/jvm/master_spec.rb

```diff
@@ -56,4 +56,41 @@ def shutdown()
       expect(Puppet.settings).to be_a_kind_of(Puppet::Server::Settings)
     end
   end
+
+  context "#check_ca_dir_for_deprecation_warning" do
+    before do
+      @master = Puppet::Server::Master.new({}, {})
+      @ssldir = Puppet[:ssldir]
+      @cadir = File.join(@ssldir, 'ca')
+    end
+
+    it "warns when there is a directory in puppet's ssl dir" do
+      expect(File).to receive(:exist?).with(@cadir).and_return(true)
+      expect(File).to receive(:symlink?).with(@cadir).and_return(false)
```

Comment:
Should these be `allow`s instead of `expect`s? I'm rusty on my rspec, I think both might work, but when I see `expect`, I think "this is the thing being tested", as opposed to `allow`, which is used for mocking? right? 
Otherwise, how are these `expect`s different from the one we *actually* want to test?

Reply:
The difference between `allow` and `expect` is that tests fail if the expectation is not met, while tests will pass if the allowance does not occur. I think it's clearer what the logic should be with `expect`, so that you aren't possibly introducing any side effects you didn't anticipate.

Reply:
hmm. I guess then I'm confused about what is testing vs. what is just setting up the scenario we're trying to test, if they're all `expect`s. Like, if the above^ `expect`s are just mocking a directory, then how is the below `expect` not just also mocking? And if none of them are mocking, then how are we setting up this specific scenario to test?
Am I making any sense?

Reply:
The `and_return` is the part that does the mocking. Without it, it just tests that the thing is called and lets it return whatever it really returns. I'm fine with these being `expect`.